### PR TITLE
Add ability to parse static functions

### PR DESCRIPTION
**Pull request by kant2002**
Most implementation was simply refactoring.

Basically I want make operation closer to how C perceived by developer.
- We declare function if possible. We can declare multiple times. All of that operations happens in Lower phase. This allow combine various properties like storage class until emit happens. CLI import resolved at this stage too.
- Then we define method definition for the function. We simply create new method for all definitions.
- Then emit body (this is not changed)

### CODE REVIEW

File: Cesium.IntegrationTests/function_storage.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    printf("inside storage_declared_worker");
+}
+
+/*
+void storage_defined_worker();
+
+static void storage_defined_worker()
+{
+    printf("inside storage_defined_worker");
+}*/
```
Comment (ForNeVeR) [inline Cesium.IntegrationTests/function_storage.c:23]: Why is this commented-out? Doesn't it work?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: Cesium.IntegrationTests/function_storage.c (hunk 1 above)
Comment (kant2002) [reply Cesium.IntegrationTests/function_storage.c:23]: On MSVC it’s fine. In Cesium too.

Comment (ForNeVeR) [reply Cesium.IntegrationTests/function_storage.c:23]: I suspect it breaks the standard then, let's git rid of this particular test?

Perhaps there's no sense in declaring a function with no `static` linkage, and then marking an implementation as `static`.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**